### DISCUSSION TITLE
Call a function if an actor dies

### DISCUSSION THREAD

**Discussion by ghost** [jodal/pykka#70] [Q&A]
I don't get the use of on_failure function

``` python
import pykka
class ExceptionLaunch(pykka.ThreadingActor):
    def __init__(self):
        super(ExceptionLaunch, self).__init__()
        self.pippo = {'ciccio': None}

    def on_start(self):
        logger.info('Start')

    def on_stop(self):
        logger.info('Start')

    def on_failure(self, exception_type, exception_value, traceback):
        logger.info('%s %s %s %s', exception_type, exception_value, traceback)

    def throw_exc(self):
        logger.info('Throw exception')
        logger.info('%s', self.pippo['pluto'])

 exc = ExceptionLaunch.start()
result = exc.proxy().throw_exc()
# result.get()

```

How to catch exception in an actor to signal for example is parent that the actor is dead or to log
a debug message???

in the case above if i call result.get() it throwns me an exception but if I want to call it asynchronously
the actor "die" without telling me and future messages are ignored (if i persist calling the throw_exc function it does't nothing)

am I missing something??
there's a method for example if I have a parent actor and a child actor to signal the parent that the child is died???

``` python
def on_exception(error_message):
    self._parent.proxy().child_has_died().get()
```

**Comment by tylerhjones** [comment]
Is there an answer or up to date example on handling actor failures?
